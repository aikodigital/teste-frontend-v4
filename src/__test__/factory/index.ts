export * from './Equipment';
export * from './hooks';
