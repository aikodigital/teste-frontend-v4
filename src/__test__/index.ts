export * from './sutMakeProvider';
export * from './factory';
export * from './utils';
