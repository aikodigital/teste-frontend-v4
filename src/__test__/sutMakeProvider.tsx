import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EquipmentProvider } from '@/hooks';

import { theme } from '@/styles';

export const sutMockProvider = (children: JSX.Element) => {
  return render(
    <ThemeProvider theme={theme}>
      <EquipmentProvider>{children}</EquipmentProvider>
    </ThemeProvider>
  );
};
