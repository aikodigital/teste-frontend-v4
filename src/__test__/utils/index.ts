export * from './populateInput';
export * from './populateSelect';
