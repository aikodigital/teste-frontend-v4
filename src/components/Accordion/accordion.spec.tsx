import { cleanup, fireEvent } from '@testing-library/react';
import {
  AccordionContainer,
  AccordionContent,
  AccordionItem,
  AccordionTrigger,
} from '.';
import { sutMockProvider } from '@/__test__';

describe('Accordion', () => {
  afterEach(cleanup);

  const makeTemplate = () => {
    return (
      <AccordionContainer type="single" data-testid="accordion" collapsible>
        <AccordionItem value="item-1" data-testid="accordion-item">
          <AccordionTrigger data-testid="accordion-trigger">
            <span>Accordion Title</span>
          </AccordionTrigger>

          <AccordionContent data-testid="accordion-content">
            <span>Accordion Content</span>
          </AccordionContent>
        </AccordionItem>
      </AccordionContainer>
    );
  };

  const makeSut = () => {
    const sut = sutMockProvider(makeTemplate());

    return {
      sut,
    };
  };

  it('should render the component', () => {
    const { sut } = makeSut();

    expect(sut.getByTestId('accordion')).toBeInTheDocument();
  });

  it('should show the content when the trigger is clicked', () => {
    const { sut } = makeSut();

    const trigger = sut.getByText('Accordion Title');

    fireEvent.click(trigger);

    const content = sut.getByText('Accordion Content');

    expect(content).toBeInTheDocument();
  });

  it('should hide the content when the trigger is clicked twice', () => {
    const { sut } = makeSut();

    const trigger = sut.getByText('Accordion Title');

    fireEvent.click(trigger);
    fireEvent.click(trigger);

    const content = sut.queryByText('Accordion Content');

    expect(content).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { sut } = makeSut();

    expect(sut.container).toMatchSnapshot();
  });
});
