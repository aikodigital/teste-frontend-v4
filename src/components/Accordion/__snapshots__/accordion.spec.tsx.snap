// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion > should match snapshot 1`] = `
<div>
  <div
    class="sc-braxZu gaSGXv"
    data-orientation="vertical"
    data-testid="accordion"
  >
    <div
      class="sc-gJhJTp iaxenZ"
      data-orientation="vertical"
      data-state="closed"
      data-testid="accordion-item"
    >
      <h3
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r7:"
          aria-expanded="false"
          class="sc-elDIKY bBOVoC"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger"
          id="radix-:r6:"
          type="button"
        >
          <span>
            Accordion Title
          </span>
          <svg
            aria-hidden="true"
            class="AccordionChevron"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r6:"
        class="sc-fQpRED bYTzAp"
        data-orientation="vertical"
        data-state="closed"
        data-testid="accordion-content"
        hidden=""
        id="radix-:r7:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
